api/v1alpha3/vspherefailuredomain_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VSphereFailureDomainSpec defines the desired state of VSphereFailureDomain
+type VSphereFailureDomainSpec struct {
+
+	// Region defines the name and type of a region
+	Region FailureDomain `json:"region"`
+
+	// Zone defines the name and type of a zone
+	Zone FailureDomain `json:"zone"`
+
+	// Topology is the what describes a given failure domain using vSphere constructs
+	Topology Topology `json:"topology"`
+}
+
+type FailureDomainType string
+
+type FailureDomain struct {
+	// Name is the name of the tag that represents this failure domain
+	Name string `json:"name"`
+
+	// Type is the type of failure domain, the current values are "Datacenter", "ComputeCluster" and "HostGroup"
+	Type FailureDomainType `json:"type"`
```

Comment:
similar to the identity type that I'm implementing. Should we put an enum validation for the types that it could be? `+kubebuilder:validation:Enum=Datacenter;ComputeCluster;HostGroup`